servant-server/src/Servant/Server/Internal.hs

```diff
@@ -880,5 +882,31 @@ type HasServerArrowTypeError a b =
     ':$$: 'ShowType b
 #endif
 
+-- | Ignore @'Fragment'@ in server handlers.
+-- See <https://ietf.org/rfc/rfc2616.html#section-15.1.3> for more details.
+--
+-- Example:
+--
+-- > type MyApi = "books" :> Fragment Text :> Get '[JSON] [Book]
+-- >
+-- > server :: Server MyApi
+-- > server = getBooksBy
+-- >   where getBooksBy :: Handler [Book]
+-- >         getBooksBy = ...return all books...
+#ifdef HAS_TYPE_ERROR
+instance ( OnlyOneFragment api, HasServer api context, FromHttpApiData a1
+         , SBoolI (FoldLenient mods)
+         )
+#else
+instance (HasServer api context, FromHttpApiData a1, SBoolI (FoldLenient mods))
```

Comment:
`FoldLenient` is not needed anymore, right?

Reply:
Fixed.